README.md

```diff
@@ -219,6 +219,12 @@ In cases where this behavior is undesirable, use the `skip_required` option:
 <%= f.password_field :password, label: "New Password", skip_required: true %>
 ```
 
+In cases where the label `required` class should be added but the associated model has no presence validation, use the `required` option:
```

Comment:
Please add a line that says the `required: false` does nothing, or, change the code so that `required: false` is equivalent to `skip_required: true`. If you change the code, don't forget to add a test case for `required: false`. If you just change the documentation, I don't think you need to add a test case.

I don't have an opinion about which is the right solution. Since you have the use case, I'm happy to leave it to your judgement.

Reply:
Good catch !

To simplify the API I would do the following : 
- deprecate skip_required
  * add a warning (like rails is doing)
  * raise an error if skip_required and the new required options are used at the same time
- allow 3 values on the new required option:
  * `:infered` : default behavior (required class is added if validation is present)
  * `false` : required class NOT added, whatever the model validation
  * `true` : required class added, whatever the model validation

Or I'll add a line in the readme to say that `required: false` does nothing.

Reply:
I love the enthusiasm. If you want to modify your PR for what you listed under "To simplify the API...", it sounds good to me. 

The only suggestion I have is that we make it a warning rather than an error if both `skip_required` and `required` are used, and use the `skip_required` option. This would prevent breaking sites where someone in the past:

1. Used `required` without looking in the docs.
2. When it didn't work, saw that the option was `skip_required`.
3. Added `skip_required` but forgot to remove the `required` option.

I know that's a bit extreme, but I think this case warrants a warning rather than raising an exception.

One small detail: "inferred" has two "r". (And no need to apologize. English spelling sucks. I had to look it up myself to be sure.) Or you could just use `:default` instead of `inferred`. I think some other options use `:default`.